### ISSUE TITLE
Functions with multiple arguments, can't be memoized ?

### ISSUE THREAD

**Issue by marcoarthur** [r-lib/memoise#94]
I don't know, but I can't get working with functions with multiple args: I just tried from the example section, and got this:
<pre>
> b <- function(n, dummy="a") { runif(n) }
> memB <- memoise(b)
> memB(2)
[1] 0.36010744 0.07693507
> formals(memB)
$x

> formals(b)
$n

$dummy
[1] "a"

> 
</pre>

In manual it says the formals should be the same, but here we see memoized function had one argument only

**Comment by cderv** [comment]
FWIWI, I got the same formals with the example. Here is a reprex
``` r
library(memoise)
packageVersion("memoise")
#> [1] '1.1.0'
b <- function(n, dummy="a") { runif(n) }
memB <- memoise(b)

formals(b)
#> $n
#> 
#> 
#> $dummy
#> [1] "a"
formals(memB)
#> $n
#> 
#> 
#> $dummy
#> [1] "a"

memB(2)
#> [1] 0.8866823 0.5315909
memB(2, dummy = "a")
#> [1] 0.8866823 0.5315909
replicate(5, b(2))
#>           [,1]      [,2]      [,3]      [,4]       [,5]
#> [1,] 0.3744030 0.1949030 0.3893541 0.6695931 0.07990824
#> [2,] 0.8302855 0.3858364 0.2920963 0.9717740 0.21338588
replicate(5, memB(2))
#>           [,1]      [,2]      [,3]      [,4]      [,5]
#> [1,] 0.8866823 0.8866823 0.8866823 0.8866823 0.8866823
#> [2,] 0.5315909 0.5315909 0.5315909 0.5315909 0.5315909
```

<sup>Created on 2019-08-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by marcoarthur** [comment]
Can't use reprex (well don't know how) in a remote Rstudio (web browser based) that I'm using. After scratch my head I found the problem. 

I just had a function the same name, I copied from [here](https://www.r-bloggers.com/optimize-your-r-code-using-memoization/) and that function only accepts one argument function as input, that was the problem. Thanks your help, now I found that reprex exist, really nice.
